clock_exercise/CheckBox.qml

```diff
@@ -0,0 +1,25 @@
+import QtQuick 2.0
+
+MouseArea {
+    id: root
+
+    property bool on: false
+
+    signal changed
+
+    width: 28
+    height: 29
+    onClicked: changed()
```

Comment:
Qui c'è anche la possibilità di usare un approccio diverso, ovvero di gestire internamente il cambio di stato grafico avendo:
```qml
onClicked: { on = !on }
```
Impostato così come adesso significa che questa componente richiederà SEMPRE un intervento esterno per cambiare il suo stato grafico.

Entrambi gli approcci (quello col cambio di stato interno e quello pilotato da fuori) hanno i loro PRO e CONTRO.

L'approccio del cambio di stato gestito internamente alla componente è molto più comodo se hai a che fare con cambi di stato sincroni, ovvero se al cambio di stato della checkbox noi siamo in grado di cambiare immediatamente lo stato della cosa a cui la checkbox si riferisce.

Prendiamo ora ad esempio il caso di una checkbox che serve per accendere/spegnere una luce smart in casa: usando l'approccio interno, al cambio di stato grafico della checkbox (che è istantaneo) non corrisponderebbe il cambio di stato della lampadina (perchè ci sono le latenze introdotte dalla comunicazione al dispositivo); in più l'approccio interno non è molto fault-tolerant, ad esempio se la lampadina fosse fulminata allora la checkbox continuerebbe a mostrare un cambio di stato anche quando in realtà non c'è nessuna accensione. 

L'approccio del cambio di stato pilotato dall'esterno della componente invece è molto più comodo nei casi asincroni come quello della lampadina, perchè controllando la grafica da fuori hai pieno controllo su come gestire le varie casistiche.

Reply:
Si era inutile fare il cambio da fuori, inoltre c'era una property di troppo, ho fatto che la CheckBox agisce direttamente su selected ed ho eliminato la property on